src/backend/fs.rs

```diff
@@ -162,11 +168,12 @@ pub fn write_catalog(file_hash: &Blake3Hash, segment_hashes: &[BaoHash]) -> Resu
             let path = volume.path.join(CATALOG_DIR).join(file_hash.to_string());
 
             trace!("Open catalog file at {}", path.to_string_lossy());
-            let mut file = OpenOptions::new()
-                .create_new(true)
-                .write(true)
-                .open(&path)?;
+            let mut file = fs::File::create(&path)?;
 
+            // let mut file = OpenOptions::new()
+            //     .create_new(true)
+            //     .write(true)
+            //     .open(&path)?;
```

Comment:
This solved an Internal Error 500 while using multi-part.
Good to point out it was over-writing the catalog. 
This change has no value since we are not even using multi-part.